app/models/concerns/language_and_schedule_references.rb

```diff
@@ -5,6 +5,20 @@ module LanguageAndScheduleReferences
     has_many :language_skills, as: :languageable, dependent: :destroy
     accepts_nested_attributes_for :language_skills, allow_destroy: true
 
+    def first_language
```

Comment:
> True, there is a naming conflict with rails/ruby naming conventions.

Well it's not ruby or rails specific, the name `first_language` is singular and contains the word `first`, so it clearly implies that the method returns the first item only.

> I'll make it a native_language then, if that suits better...

`native_language **s** (-> plural)

Also, you're calling it as `volunteer.first_language.first`, so the method can just as well directly return 1 item.